docs/introduction/migrate-from-v7.md

```
@@ -0,0 +1,598 @@
+## Switch your code base on ESM
+
+In v8, Ts.ED only supports ECMAScript Modules (ESM). You have to switch your code base to ESM.
+This task can be complicated depending on your project structure.
+
+::: tip
+v7 is already compatible with ESM, so you can start to switch your code base to ESM in v7.
+:::
+
+Here are some steps to help you to switch your code base to ESM.
+
+#### add `.js` extension in your imports statements.
```

Markdownlint

Expected: h3; Actual: h4
Heading levels should only increment by one level at a time

(MD001, heading-increment)